app/scripts/modules/core/config/settings.ts

```diff
@@ -0,0 +1,174 @@
+    namespace: string;
+    proxy: string;
+  };
+}
+
+interface IAppengineProviderSettings {
+  defaults: {
+    account: string;
+    editLoadBalancerStageEnabled: boolean;
+  };
+}
+
+
+interface IProviderSettings {
+  azure?: IAzureProviderSettings;
+  aws?: IAWSProviderSettings;
+  gce?: IGCEProviderSettings;
+  titus?: ITitusProviderSettings;
+  openstack?: IOpenStackProviderSettings;
+  kubernetes?: IKubernetesProviderSettings;
+  appengine?: IAppengineProviderSettings;
+  [key: string]: any; // allows custom providers not typed in here (good for testing too)
+}
+
+interface INotificationSettings {
+  email: {
+    enabled: boolean;
+  };
+  hipchat: {
+    enabled: boolean;
+    botName: string;
+  };
+  sms: {
+    enabled: boolean;
+  };
+  slack: {
+    enabled: boolean;
+    botName: string;
+  };
+}
+
+export interface ISpinnakerSettings {
+  checkForUpdates: boolean;
+  debugEnabled: boolean;
+  defaultProviders: string[];
+  feedbackUrl: string;
+  gateUrl: string;
+  bakeryDetailUrl: string;
+  authEndpoint: string;
+  pollSchedule: number;
+  defaultTimeZone: string; // see http://momentjs.com/timezone/docs/#/data-utilities/
+  defaultCategory: string;
+  defaultInstancePort: number;
+  providers?: IProviderSettings;
+  whatsNew: {
+    gistId: string;
+    fileName: string;
+    accessToken?: string;
+  };
+  notifications: INotificationSettings;
+  authEnabled: boolean;
+  authTtl: number;
+  gitSources: string[];
+  triggerTypes: string[];
+  analytics: {
+    ga?: boolean;
+  };
+  feature: {
+    entityTags?: boolean;
+    fiatEnabled?: boolean;
+    pipelines?: boolean;
+    notifications?: boolean;
+    fastProperty?: boolean;
+    vpcMigrator?: boolean;
+    clusterDiff?: boolean;
+    roscoMode?: boolean;
+    netflixMode?: boolean;
+    chaosMonkey?: boolean;
+    // whether stages affecting infrastructure (like "Create Load Balancer") should be enabled or not
+    infrastructureStages?: boolean;
+    jobs?: boolean;
+    snapshots?: boolean;
+    dockerBake?: boolean;
+    blesk?: boolean;
+    tableau?: boolean;
+    [key: string]: any;
+  };
+  feedback?: {
+    slack?: {
+      team: string;
+      teamName: string;
+      helpChannel: string;
+      helpChannelName: string;
+    }
+  };
+  chap?: {
+    canaryReportBaseUrl: string;
+    chapBaseUrl: string;
+    fitBaseUrl: string;
+  };
+  tableau?: {
+    appSourceUrl: string;
+    summarySourceUrl: string;
+  };
+  executionWindow?: {
+    atlas?: {
+      regions: { label: string, baseUrl: string }[];
+      url: string;
+    }
+  };
+  entityTags?: {
+    maxUrlLength?: number;
+  };
+  [key: string]: any;
+}
+
+export const SETTINGS: ISpinnakerSettings = (<any>window).spinnakerSettings;
+
+// Make sure to set up some reasonable default settings fields so we do not have to keep checking if they exist everywhere
+SETTINGS.feature = SETTINGS.feature || {};
+SETTINGS.analytics = SETTINGS.analytics || {};
+SETTINGS.providers = SETTINGS.providers || {};
```

Comment:
It might be nice to add a couple of bits here to make test setup/cleanup easier:
```
const originalSettings: ISpinnakerSettings = _.cloneDeep(SETTINGS);
SETTINGS.afterTest = () => { _.merge(SETTINGS, originalSettings); }
```

Then, in tests, you can just call:
```javascript
afterEach(SETTINGS.afterTest);
```